### PR TITLE
Added case to verify invalid syntax when using private field on object destructuring

### LINKED ISSUE #2072: Missing private field access on object destructuring
It needs to throw a SyntaxError:

```
class Foo {
    #x = 1;
    destructureX() {
        const { #x: x } = this;
        return x;
    }
}
```

### PR DESCRIPTION
**Pull request by caiolima**
This closes #2072.

### CODE REVIEW

File: src/class-elements/grammar-private-field-on-object-destructuring.case [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+info: |
+  Updated Productions
+
+  ObjectAssignmentPattern[Yield, Await]:
+     {}
+     {AssignmentRestProperty[?Yield, ?Await]}
+     {AssignmentPropertyList[?Yield, ?Await]}
+     {AssignmentPropertyList[?Yield, ?Await],AssignmentRestProperty[?Yield, ?Await]opt}
+template: syntax/invalid
+features: [class-fields-private]
```
Comment (caitp) [inline src/class-elements/grammar-private-field-on-object-destructuring.case]: add `destructuring-binding`

File: test/language/expressions/class/elements/syntax/early-errors/grammar-private-field-on-object-destructuring.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// This file was procedurally generated from the following sources:
+// - src/class-elements/grammar-private-field-on-object-destructuring.case
+// - src/class-elements/syntax/invalid/cls-expr-elements-invalid-syntax.template
+/*---
+description: Acessing private field from object destructuring pattern is not a valid syntax (class expression)
+esid: prod-ClassElement
+features: [class-fields-private, class]
```
Comment (caitp) [inline test/language/expressions/class/elements/syntax/early-errors/grammar-private-field-on-object-destructuring.js]: ditto

File: test/language/statements/class/elements/syntax/early-errors/grammar-private-field-on-object-destructuring.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// This file was procedurally generated from the following sources:
+// - src/class-elements/grammar-private-field-on-object-destructuring.case
+// - src/class-elements/syntax/invalid/cls-decl-elements-invalid-syntax.template
+/*---
+description: Acessing private field from object destructuring pattern is not a valid syntax (class declaration)
+esid: prod-ClassElement
+features: [class-fields-private, class]
```
Comment (caitp) [inline test/language/statements/class/elements/syntax/early-errors/grammar-private-field-on-object-destructuring.js]: ditto

File: src/class-elements/grammar-private-field-on-object-destructuring.case [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+---*/
+
+//- elements
+#x = 1;
+
+destructureX() {
+  const { #x: x } = this;
+  return x;
```
Comment (leobalter) [inline src/class-elements/grammar-private-field-on-object-destructuring.case]: nit: while you're working on the proposed changes, this line is not necessary and can be removed.

**Comment by leobalter** [comment]
awaiting for changes requested by @caitp.

File: src/class-elements/grammar-private-field-on-object-destructuring.case (hunk 1 above)
Comment (caiolima) [reply src/class-elements/grammar-private-field-on-object-destructuring.case]: Done!

**Pull request merged**